### PR TITLE
Require docs for public members

### PR DESCRIPTION
**Pull request by jayvdb**

### CODE REVIEW

File: progenitor-client/src/progenitor_client.rs [hunk 1]
```diff
@@ -386,7 +386,7 @@ const PATH_SET: &percent_encoding::AsciiSet = &percent_encoding::CONTROLS
     .add(b'/')
     .add(b'%');
 
-#[doc(hidden)]
+/// Percent encode input string.
```
Comment (ahl) [inline progenitor-client/src/progenitor_client.rs:389]: why doc this? this is really only for internal use.

File: progenitor-client/src/lib.rs [hunk 2]
```diff
@@ -1,5 +1,9 @@
 // Copyright 2022 Oxide Computer Company
 
+//! Client support for the progenitor OpenAPI client generator.
```
Comment (ahl) [inline progenitor-client/src/lib.rs:3]: What do you think?

```suggestion
//! Support for generated clients.
```

File: progenitor-client/src/progenitor_client.rs (hunk 1 above)
Comment (jayvdb) [reply progenitor-client/src/progenitor_client.rs:389]: It is part of the public API.

Comment (ahl) [reply progenitor-client/src/progenitor_client.rs:389]: Ok: I don't want to document this. I don't want people using it explicitly.

Comment (jayvdb) [reply progenitor-client/src/progenitor_client.rs:389]: ok, ill revert this bit

**Review by ahl — APPROVED** [review]

**Comment by jayvdb** [comment]
@ahl , PR refreshed & green.

**Pull request merged**